private/sync2/tee.go

```diff
@@ -31,6 +31,7 @@ func NewTeeFile(readers int, tempdir string) ([]PipeReader, PipeWriter, error) {
 	if err != nil {
 		return nil, nil, err
 	}
+	os.Remove(tempfile)
```

Comment:
How does this work? If we remove the temp file immediately after creating it, how is it going to be used as a memory buffer?!

Reply:
Also, there is no error check.

Reply:
https://pubs.opengroup.org/onlinepubs/9699919799/functions/unlink.html

> When the file's link count becomes 0 and no process has the file open, the space occupied by the file shall be freed and the file shall no longer be accessible. If one or more processes have the file open when the last link is removed, the link shall be removed before unlink() returns, but the removal of the file contents shall be postponed until all references to the file are closed.

The behavior is that the operating system will keep the file information (and storage) around until all open file handles have been closed (despite the directory entry for the file being removed).

Reply:
Interesting! A new thing I learned.

Reply:
Can we add this info as a comment in the code?